From 03f3e2574dc5f8cd7a1fad83edee7f1d9f6b45cb Mon Sep 17 00:00:00 2001
From: Felipe Contreras <felipe.contreras@gmail.com>
Date: Mon, 11 Apr 2011 13:41:20 +0300
Subject: [PATCH] x86_64: fix for new memcpy behavior

On 2.14 memcpy@GLIBC_2.2.5 is an alias to memmove, only
memcpy@GLIBC_2.13 uses the new behavior.

This backports the old behavior for versions before than 2.14.

Signed-off-by: Felipe Contreras <felipe.contreras@gmail.com>
---
 sysdeps/x86_64/multiarch/memcpy-ssse3-back.S |    1 +
 sysdeps/x86_64/multiarch/memcpy-ssse3.S      |    1 +
 2 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/sysdeps/x86_64/multiarch/memcpy-ssse3-back.S b/sysdeps/x86_64/multiarch/memcpy-ssse3-back.S
index 48c974e..2867d45 100644
--- a/sysdeps/x86_64/multiarch/memcpy-ssse3-back.S
+++ b/sysdeps/x86_64/multiarch/memcpy-ssse3-back.S
@@ -30,6 +30,7 @@
 #ifndef MEMCPY
 # define MEMCPY		__memcpy_ssse3_back
 # define MEMCPY_CHK	__memcpy_chk_ssse3_back
+# define USE_AS_MEMMOVE
 #endif
 
 #ifndef ALIGN
diff --git a/sysdeps/x86_64/multiarch/memcpy-ssse3.S b/sysdeps/x86_64/multiarch/memcpy-ssse3.S
index 9a878d3..fb60e06 100644
--- a/sysdeps/x86_64/multiarch/memcpy-ssse3.S
+++ b/sysdeps/x86_64/multiarch/memcpy-ssse3.S
@@ -30,6 +30,7 @@
 #ifndef MEMCPY
 # define MEMCPY		__memcpy_ssse3
 # define MEMCPY_CHK	__memcpy_chk_ssse3
+# define USE_AS_MEMMOVE
 #endif
 
 #ifndef ALIGN
-- 
1.7.5.rc1

